server/komodo-server/src/main/java/org/komodo/rest/connections/SyndesisConnectionSynchronizer.java

```diff
@@ -94,14 +94,25 @@ private void handleAddConnection(EventMsg event, boolean update) throws KExcepti
     private void handleDeleteConnection(EventMsg event) throws KException {
         // note here that the datasource is already deleted from the syndesis
         // so we would need to search by local cached event id
-        DefaultSyndesisDataSource sds = this.openshiftClient.getSyndesisDataSourceById(bogusCredentials,
-                event.getId());
+        DefaultSyndesisDataSource sds = this.openshiftClient.getSyndesisDataSourceById(event.getId());
         if (sds != null) {
             deleteConnection(sds);
         }
     }
 
     public void addConnection(DefaultSyndesisDataSource sds, boolean update) {
+        try {
+            // this is avoid circular creation of the virtualization connection that is
+            // published through syndesis
+            for (DataVirtualization dv : kengine.getWorkspaceManager().findDataVirtualizations()) {
```

Comment:
This should be replaced by a new repository method - findDataVirtualizationBySourceId, or a boolean function.